### ISSUE TITLE
LC0003 throws NullReferenceException in PreRelease

### ISSUE THREAD

**Issue by jwikman** [StefanMaron/BusinessCentral.LinterCop#506]
It seems as if #501 caused a NullReferenceException in my project 🤔

I cannot see where though, as this is thrown in app.json.

Maybe need to introduce LC0000 again, if you do not know immediately what is wrong?

```
[{
	"resource": "/d:/VSCode/Git/NAB/Eagle/App/app.json",
	"owner": "_generated_diagnostic_collection_name_#0",
	"code": "AD0001",
	"severity": 4,
	"message": "Analyzer 'BusinessCentral.LinterCop.Design.Rule0003DoNotUseObjectIDsInVariablesOrProperties' threw an exception of type 'System.NullReferenceException' with message 'System.NullReferenceException: Object reference not set to an instance of an object.\r\n   at BusinessCentral.LinterCop.Design.Rule0003DoNotUseObjectIDsInVariablesOrProperties.CheckForObjectIDsInVariablesOrProperties(SyntaxNodeAnalysisContext ctx)\r\n   at Microsoft.Dynamics.Nav.CodeAnalysis.Diagnostics.AnalyzerExecutor.<>c__DisplayClass52_1.<ExecuteSyntaxNodeAction>b__1() in D:\\a\\_work\\1\\s\\source\\Prod\\Microsoft.Dynamics.Nav.CodeAnalysis\\DiagnosticAnalyzer\\AnalyzerExecutor.cs:line 745\r\n   at Microsoft.Dynamics.Nav.CodeAnalysis.Diagnostics.AnalyzerExecutor.ExecuteAndCatchIfThrows_NoLock(DiagnosticAnalyzer analyzer, Action analyze, Nullable`1 info) in D:\\a\\_work\\1\\s\\source\\Prod\\Microsoft.Dynamics.Nav.CodeAnalysis\\DiagnosticAnalyzer\\AnalyzerExecutor.cs:line 1102'",
	"source": "AL",
	"startLineNumber": 1,
	"startColumn": 1,
	"endLineNumber": 1,
	"endColumn": 1
}]

```

**Comment by Arthurvdv** [comment]
Thank you for reporting this!

Maybe I've found the issue, but to be sure I've also (temporarily) wrapped it inside a LC0000 in the updated prelease [v0.30.12](https://github.com/StefanMaron/BusinessCentral.LinterCop/releases/tag/v0.30.12) version.

Could you verify if this makes any difference?

**Comment by jwikman** [comment]
Yes, the NullReferenceException is now caught by LC0000. :)

Both are thrown on TryFunctions, and LC0000 marks the table name "QWEEG Advanced KPI" for both:
```
    [TryFunction]
    internal procedure TryCalculateExpression(AdvancedKPI: Record "QWEEG Advanced KPI"; var Result: Decimal; Depth: Integer)
```
```
    [TryFunction]
    internal procedure TryValidateExpression(AdvancedKPI: Record "QWEEG Advanced KPI")
```

**Comment by jwikman** [comment]
Oh, I forgot to add the LC0000 problem:
```
[{
	"resource": "/d:/VSCode/Git/NAB/Eagle/App/src/KPIs/AdvancedKPIs/Expressions/ExpressionMgt.Codeunit.al",
	"owner": "_generated_diagnostic_collection_name_#0",
	"code": {
		"value": "LC0000",
		"target": {
			"$mid": 1,
			"external": "https://github.com/StefanMaron/BusinessCentral.LinterCop/wiki/LC0000",
			"path": "/StefanMaron/BusinessCentral.LinterCop/wiki/LC0000",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 2,
	"message": "There was an Error in Rule \"Rule0003/Rule0005\" of type \"NullReferenceException\" ",
	"source": "AL",
	"startLineNumber": 34,
	"startColumn": 66,
	"endLineNumber": 34,
	"endColumn": 86
}]
```

**Comment by jwikman** [comment]
It seems as if your last change made a difference!

I do not get any problems reported from this any longer. Case closed. 🙂

**Comment by Arthurvdv** [comment]
Found the culprit and resolved in the updated (pre)release [v0.30.12](https://github.com/StefanMaron/BusinessCentral.LinterCop/releases/tag/v0.30.12) version of the LinterCop.
